pkg/registry/core/service/strategy.go

```diff
@@ -383,10 +383,7 @@ func dropTypeDependentFields(newSvc *api.Service, oldSvc *api.Service) {
 }
 
 func needsClusterIP(svc *api.Service) bool {
-	if svc.Spec.Type == api.ServiceTypeExternalName {
-		return false
-	}
-	return true
+	return svc.Spec.Type != api.ServiceTypeExternalName
```

Comment:
I prefer the way it reads now, before this change , it is easier to read

I think that there is no efficiency or performance improvements, since the compiler should generate the same code